src/Npgsql/PostgresDatabaseInfo.cs

```diff
@@ -335,93 +336,28 @@ static string SanitizeForReplicationConnection(string str)
             var len = conn.ReadBuffer.ReadInt32();
             var elemtypoid = len == -1 ? 0 : uint.Parse(conn.ReadBuffer.ReadString(len), NumberFormatInfo.InvariantInfo);
 
-            switch (typtype)
-            {
-            case 'b': // Normal base type
-                var baseType = new PostgresBaseType(nspname, typname, oid);
-                byOID[baseType.OID] = baseType;
-                continue;
-
-            case 'a': // Array
-            {
-                Debug.Assert(elemtypoid > 0);
-                if (!byOID.TryGetValue(elemtypoid, out var elementPostgresType))
-                {
-                    _connectionLogger.LogTrace("Array type '{ArrayTypeName}' refers to unknown element with OID {ElementTypeOID}, skipping",
-                        typname, elemtypoid);
-                    continue;
-                }
-
-                var arrayType = new PostgresArrayType(nspname, typname, oid, elementPostgresType);
-                byOID[arrayType.OID] = arrayType;
-                continue;
-            }
+            var postgresTypeDefinition = new PostgresTypeDefinition(nspname, oid, typname, typtype, typnotnull, elemtypoid);
+            if (!TryAddPostgresType(postgresTypeDefinition, byOID))
+                unknownPostgresTypes.Add(postgresTypeDefinition);
+        }
 
-            case 'r': // Range
+        while (unknownPostgresTypes.Count > 0)
```

Comment:
I know this is paranoid/defensive, but when I see a loop like this, I find myself wanting to have a safety check to avoid an infinite loop (in case there's some sort of corrupt data or whatever where one type references another), just in case. i.e. if there are two loop iterations where unknownPostgresTypes.Count hasn't changed, throw...

Reply:
There is one already? At least that's what "hasChanges" is supposed to do...

Reply:
Right... Is there an actual scenario where we expect this to happen? Am just wondering if it's better to throw rather than to break (e.g. this would cause us to skip later types, better not to hide the problem maybe?)

Reply:
In vanilla postgres? Most likely not. At least I can't imagine a scenario where we could have a type referencing another type by we somehow never get it. In a postgres fork? Who knows, we get reports from time to time of some weird behavior, like for the example the last issue, where yugabytedb doesn't return RowDescription for DDL queries. Overall I do not think there is a lot of reasons to be extremely strict regarding loading types. In the worst case scenario users will get an error saying that a particular type couldn't be mapped and afterwards they can enable logging, which will state that we couldn't find the base type. And given that it's probably only gonna happen with some internal type which no user will ever use, eh.

Reply:
OK, am OK to go in this direction. Maybe consider doing a warning for this case though?